app/assets/javascripts/lookup_keys.js

```diff
@@ -215,3 +221,60 @@ function validatorTypeSelected(item){
   var validator_rule_field = $(item).closest('.fields').find("[id$='_validator_rule']");
   validator_rule_field.attr('disabled', validatorType == "" ? 'disabled' : null);
 }
+
+const KEY_DELM = ",";
+const EQ_DELM  = "=";
+
+function match_to_key_value(match){
+  var keys = [], values = [],
+    split_matcher = match.replace(/ /g,'').split(/[,=]/);
+
+  $.each(split_matcher, function (index, value) {
+    if (index % 2 === 0) {
+      keys.push(value);
+    } else {
+      values.push(value);
+    }
+  });
+
+  return [keys.join(KEY_DELM), values.join(KEY_DELM)];
+}
+
+function key_value_to_match(keys, values){
+  var match = "";
+  keys.split(KEY_DELM).forEach(function (el, index) {
+   match += el + EQ_DELM + values.split(KEY_DELM)[index] + KEY_DELM;
+  });
+
+  return match.slice(0, -1);
+}
+
+function fill_in_matchers(){
+  $('.matchers').each(function () {
+    var matcher = $(this);
+    var match = matcher.find('.match').val();
+    var matcher_key = matcher.find('.matcher_key');
+    var matcher_value = matcher.find('.matcher_value');
+    var order = matcher.closest('.matcher-parent').find('#order').val().split('\n');
+    matcher_key.empty();
+    matcher_key.append("<option></option>");
```

Comment:
Not sure I follow.  If it really can't be removed then you need to prevent the form from being saved when the matcher is set to the blank value - I was able to save one with "blank = Hostgroup".